components/formats-gpl/src/loci/formats/in/OlympusINIReader.java

```diff
@@ -0,0 +1,950 @@
+
+      for (int i=0; i<currIniConf.totalTiles; i++)
+      {
+        double xPixel;
+
+        xPixel=(double)(xMax - currIniConf.xyArr.get(i).x)/(double)currIniConf.xAdj;
+        currIniConf.xyArr.get(i).xPixel=(int)xPixel;
+        
+        double yPixel;
+        yPixel=(double)(yMax - currIniConf.xyArr.get(i).y)/(double)currIniConf.yAdj;
+        currIniConf.xyArr.get(i).yPixel=(int)yPixel;
+
+        LOGGER.debug("filename=" + currIniConf.xyArr.get(i).fileName + " x=" + xPixel + " y=" + yPixel);
+      }
+      Collections.sort(currIniConf.xyArr, new Comparator<JpegFileXY>() {
+        @Override
+        public int compare(JpegFileXY coord1, JpegFileXY coord2) {
+          int result = coord1.yPixel - coord2.yPixel;
+          if (result == 0) result = coord1.xPixel - coord2.xPixel;
+          return result;
+        }
+      });
+    }
+   
+    Collections.sort(iniConf, new Comparator<IniConf>() {
+      @Override
+      public int compare(IniConf conf1, IniConf conf2) {
+        long result = (long)(conf1.totalWidth * conf1.totalHeight) - (long)(conf2.totalWidth * conf2.totalHeight);
+        if (result > 0) return -1;
+        else if (result < 0) return 1;
+        else return 0;
+      }
+    });
+
+    baseWidth = iniConf.get(0).totalWidth;
+    baseHeight = iniConf.get(0).totalHeight;
+    LOGGER.debug("Total number of iniConf: " + iniConf.size());
+    core.clear();
+    for (int i = 0; i < 4; i++) {
+      if (iniConf.get(i).found) {
+       core.add(new CoreMetadata());
+      }
+    }
+    int totalCores = 0;
+    for (int i = 0; i < 4; i++) {
+      if (iniConf.get(i).found) {
+        CoreMetadata m = core.get(totalCores);
+        m.sizeX = (int) iniConf.get(i).totalWidth;
+        m.sizeY = (int) iniConf.get(i).totalHeight;
+        m.sizeZ = 1;
+        m.sizeC = 3;
+        m.sizeT = 1;
+        m.rgb = true;
+        m.imageCount = 1;
+        m.dimensionOrder = "XYCZT";
+        m.pixelType = FormatTools.UINT8;
+        m.interleaved = true;
+        m.falseColor = false;
+        m.thumbnail = false;
+        m.orderCertain = true;
+        m.bitsPerPixel = 8;
+        m.thumbSizeX = 0;
+        m.thumbSizeY = 0;
+        for (int j = i+1; j < 4; j++) {
+          if (iniConf.get(j).found) {
+            m.thumbSizeX = (int) iniConf.get(j).totalWidth;
+            m.thumbSizeY = (int) iniConf.get(j).totalHeight;
+          }
+        }
+        m.littleEndian = false;
+        m.indexed = false;
+        if (i==0 && totalLevels > 2) 
+        {
+          //m.resolutionCount = 1;
+          m.resolutionCount = totalLevels;
+        }
+        else 
+        {
+          m.thumbnail = true;
+          m.resolutionCount = 1;
+        }
+        totalCores++;
+      }
+    }
+    MetadataStore store = makeFilterMetadata();
+    MetadataTools.populatePixels(store, this);
+
+    String path = new Location(currentId).getAbsoluteFile().getName();
+
+    if (totalLevels > 3)
```

Comment:
For this series of if statements settings the ImageName, it is always creating 4 image objects regardless of if the resolutions are being flattened or not. Here you will also need to use the getSeriesCount() to only set ImageName on Images upto the expected series length.